erpnext/accounts/doctype/pos_invoice/pos_invoice.py

```diff
@@ -212,6 +213,26 @@ def validate_serialised_or_batched_item(self):
 		if error_msg:
 			frappe.throw(error_msg, title=_("Invalid Item"), as_list=True)
 
+	def validate_serial_nos(self):
+		from erpnext.stock.doctype.serial_no.serial_no import get_serial_nos
+		error_msg = []
+		for item in self.get("items"):
+			serialized = item.get("has_serial_no")
+			msg = ""
+			if serialized:
+				valid_serial_nos = frappe.get_all('Serial No', filters={'item_code':item.get("item_code"), "status":'Active'}, pluck="name")
```

Comment:
This should be fetched using `auto_fetch_serial_number` since it takes into account the `pos_reserved_serial_nos`